### PR TITLE
Add opt-in to opt-out line

### PR DESCRIPTION
**Pull request by gsvr**

### CODE REVIEW

**Comment by imsickofmaps** [comment]
screenshot 2014-08-18 17 23 48

**Comment by gsvr** [comment]
@imsickofmaps Review please.

File: src/utils.js [hunk 1]
```diff
@@ -792,7 +792,7 @@ go.utils = {
         });
     },
 
-    subscription_unsubscribe_all: function(contact, im, opts) {
+    subscription_unsubscribe_all: function(contact, im) {
```
Comment (imsickofmaps) [inline src/utils.js:795]: did you mean to change this?

Comment (gsvr) [inline src/utils.js:795]: Yes, it was faulty.  Didn't use opts in the function and only passed it in smsinbound.  Weird thing was when running all the tests it found opts somewhere, but when only running optout tests it failed - opts undefined.

File: src/optout.js [hunk 2]
```diff
@@ -81,6 +90,8 @@ go.app = function() {
                             .subscription_unsubscribe_all(self.contact, self.im, opts)
                             .then(function() {
                                 return Q.all([
+                                    // ensure user is not opted out
+                                    go.utils.opt_in(self.im, self.contact),
                                     // Registration is sent to optout endpoint at Jembi to indicate removal
                                     go.utils.jembi_send_json(self.contact, self.contact, 'subscription', self.im, self.metric_prefix),
```
Comment (imsickofmaps) [inline src/optout.js:93]: I've just seen that this call actually needs making for the `else` statement too. Can you add that?

File: src/optout.js [hunk 3]
```diff
@@ -27,34 +27,43 @@ go.app = function() {
...
+                                    new Choice('other', $('Other'))
+                                ],
+
+                                next: function(choice) {
+                                    self.contact.extra.opt_out_reason = choice.value;
+
+                                    return self.im.contacts
+                                        .save(self.contact)
+                                        .then(function() {
+                                            //TODO: run unsub
```
Comment (imsickofmaps) [inline src/optout.js]: remove this now, fixed in the other issue

File: src/utils.js (hunk 1 above)
Comment (imsickofmaps) [inline src/utils.js:795]: OK - thanks

**Comment by imsickofmaps** [comment]
:+1: when Travis happy

**Pull request merged**